src/java.base/share/classes/java/lang/StringLatin1.java

```diff
@@ -824,6 +825,31 @@ static Stream<String> lines(byte[] value) {
         return StreamSupport.stream(LinesSpliterator.spliterator(value), false);
     }
 
+    static void putCharsAt(byte[] val, int index, int c1, int c2, int c3, int c4) {
+        assert index >= 0 && index + 3 < length(val) : "Trusted caller missed bounds check";
+        // Don't use the putChar method, Its instrinsic will cause C2 unable to combining values into larger stores.
+        long address  = Unsafe.ARRAY_BYTE_BASE_OFFSET + index;
+        Unsafe UNSAFE = Unsafe.getUnsafe();
```

Comment:
Perhaps better to put in a `private static final` field

Reply:
already using `private static final UNSAFE`